apps/openmw/mwworld/cellstore.cpp

```diff
@@ -179,8 +182,110 @@ namespace MWWorld
         return (ref.mRef.mRefnum == pRefnum);
     }
 
+    void CellStore::moveFrom(const Ptr &object, CellStore *from)
+    {
+        mHasState = true;
+        MovedRefTracker::iterator found = mMovedToAnotherCell.find(object.getBase());
+        if (found != mMovedToAnotherCell.end())
+        {
+            // A cell we had previously moved an object to is returning it to us.
+            assert (found->second == from);
+            mMovedToAnotherCell.erase(found);
+        }
+        else
+        {
+            mMovedHere.insert(std::make_pair(object.getBase(), from));
+        }
+
+        if (mState == State_Loaded)
```

Comment:
Would it be worth to give the CellStore a reference to the World::ESMStore and the esm reader vector as a member variables? The size increase of the CellStore class should be minimal and it would simplify the interface a bit.